hivemind/server/expert_backend.py

```diff
@@ -121,18 +135,54 @@ def backward(self, *inputs: torch.Tensor) -> Tuple[torch.Tensor, ...]:
                 nested_flatten(grad_outputs), outputs_flat))
             torch.autograd.backward(outputs_flat, grad_tensors=grad_outputs_flat,
                                     create_graph=False, retain_graph=False)
-            self.apply_gradients()
+            self.apply_gradients(batch_size)
 
         return tuple(x.grad if isinstance(x.grad, torch.Tensor) else torch.zeros_like(x)
                      for x in nested_flatten((args, kwargs)))
 
-    def apply_gradients(self) -> None:
+    def apply_gradients(self, batch_size) -> None:
         """
         Train the expert for one step. This method is called by ``ExpertBackend.backward`` after computing gradients.
         """
-        self.opt.step()
-        self.opt.zero_grad()
+        # TODO support gradient accumulation
```

Comment:
[suggestion] please remove this TODO and create an issue before merging

Reply:
Created https://github.com/learning-at-home/hivemind/issues/198